lib/binary_tree.rb

```diff
@@ -0,0 +1,65 @@
+
+Node = Struct.new(:left, :right, :data)
+
+class BinaryTree
+
+attr_accessor :arr, :root
+
+  def initialize(arr)
+    @arr = arr
+    create_tree
+  end
+
+  def create_tree
+    @arr.each do |val|
+      add_node(val)
+    end
+  end
+
+  # To add the first node
```

Comment:
All methods in the class except `initialize` and `print_tree` should be private.